packages/Kokkos/CMakeLists.txt

```diff
@@ -1,6 +1,58 @@
+INCLUDE(CombinedOption)
+INCLUDE(TribitsETISupport)
+
 TRIBITS_SUBPACKAGE(Kokkos)
 
-#ADD_SUBDIRECTORY(src)
+TRIBITS_ADD_EXPLICIT_INSTANTIATION_OPTION()
+
+TRIBITS_ADD_ETI_SUPPORT()
+
+ASSERT_DEFINED(${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION)
+
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  DataTransferKit_ENABLE_EXPLICIT_INSTANTIATION
+  HAVE_DATATRANSFERKIT_EXPLICIT_INSTANTIATION
+  "Enable explicit template instantiation (ETI) in DTK"
+  ${${PACKAGE_NAME}_ENABLE_EXPLICIT_INSTANTIATION}
+  )
+
+
+#
+# Explicit template instantiation (ETI) and test instantiation logic
+#
+
+# tpetra/CMakeLists.txt (the package's CMake logic) defines these
+# variables.  Despite "ETI" in their names, they exist whether or not
+# ETI is defined.  If ETI is defined, these variables govern the set
+# of template parameter combinations over which DataTransferKit instantiates.
+# Whether or not ETI is defined, the variables govern the set of
+# template parameter combinations over which DataTransferKit runs tests.
+
+ASSERT_DEFINED (DataTransferKit_ETI_SCALARS)
+ASSERT_DEFINED (DataTransferKit_ETI_LORDS)
+ASSERT_DEFINED (DataTransferKit_ETI_GORDS)
+ASSERT_DEFINED (DataTransferKit_ETI_NODES)
```

Comment:
These get set in the top level cmakelist file so we need to use `DataTransferKit_` here. `${PACKAGE_NAME}` would give `DataTransferKitKokkos`. This is the same as setting ETI in Tpetra and TpetraCore